### PR TITLE
Azure Monitor Exporter: Basic Integration Tests

### PR DESCRIPTION
**Pull request by TimothyMothra**
## Changes
- make `OpenTelemetry.Exporter.AzureMonitor` internals visible to `Integration.Tests` project
- Add a simple test to validate that the integration test works with our Exporter
- Add internal API to mock the `AzureMonitorTransmitter`.

### CODE REVIEW

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ServiceRestClient.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 namespace OpenTelemetry.Exporter.AzureMonitor
 {
-    internal partial class ServiceRestClient
+    internal partial class ServiceRestClient : IServiceRestClient
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ServiceRestClient.cs:10]: This is an autogenerated rest client and will lose changes with autogen update. There is a way to create custom `ServiceRestClient` we may need to explore it.

Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ServiceRestClient.cs:10]: Hold that thought, i'm resolving conflicts. Looks like major changes to the RestClient.

I can sync with you offline to discuss more.
Here I'm changing a non-autogen class. 
If we can propose changes to the autogen class this won't be necessary. Please see my comment in IServiceRestClient.

**Comment by TimothyMothra** [comment]
switching this to a draft while i work on fixing the Project References

File: sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ServiceRestClient.cs (hunk 1 above)
Comment (TimothyMothra) [reply sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/src/ServiceRestClient.cs]: i abandoned this approach

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM

**Pull request merged**